### PR TITLE
replace: fix URL and hash

### PR DESCRIPTION
**Pull request by wizeman**
###### Motivation for this change

The old URL wasn't available anymore.

There is a minor version change but it's only an unimportant change to
the Makefile and some changes to the documentation.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wizeman** [comment]
Note: I haven't checked whether this breaks anything.

Also, this should be backported to 21.05.

File: pkgs/tools/text/replace/default.nix [hunk 1]
```diff
@@ -4,8 +4,8 @@ stdenv.mkDerivation {
   name = "replace-2.24";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/replace/default.nix:4]:
```suggestion
  pname = "replace";
  version = "2.24";
```

and add a rec one line above.

File: pkgs/tools/text/replace/default.nix [hunk 2]
```diff
@@ -4,8 +4,8 @@ stdenv.mkDerivation {
...
 
   src = fetchurl {
-    url = "ftp://hpux.connect.org.uk/hpux/Users/replace-2.24/replace-2.24-src-11.11.tar.gz";
-    sha256 = "1c2nkxx83vmlh1v3ib6r2xqh121gdb1rharwsimcb2h0xwc558dm";
+    url = "http://hpux.connect.org.uk/ftp/hpux/Users/replace-2.24/replace-2.24-src-11.31.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/replace/default.nix:7]:
```suggestion
    url = "http://hpux.connect.org.uk/ftp/hpux/Users/replace-2.24/replace-${version}-src-11.31.tar.gz";
```

Comment (wizeman) [reply pkgs/tools/text/replace/default.nix]: I added another `${version}` there as well. Thanks!

**Pull request merged**